docs/architecture/how/resharding.md

```diff
@@ -47,61 +66,85 @@ StateSync in this phase would send the ``StateSplitRequest`` to the ``SyncJobsAc
+
+
+## Constraints
+
+The state sync of the parent shard, the resharing and the catchup of the children shards must all complete within a single epoch. 
+
+
+## Rollout
+
+### Flow
+
+The resharding will be initiated by having it included in a dedicated protocol version together with neard . Here is the expected flow of events:
+
+* A new neard release is published and protocol version upgrade date is set to D, roughly a week from the release. 
+* All node operatores upgrade their binaries to the newly released version within the given timeframe, ideally as soon as possible but no later than D. 
+* The protocol version upgrade voting takes place at D in an epoch E and nodes vote in favour of switching to the new protocol version in epoch E+2. 
+* The resharding begins at the beginning of epoch E+1. 
+* The network switches to the new shard layout in the first block of epoch E+2. 
+
 
-That's why we cannot move it column 'in trie parts' (like we did for others), but we do it by 'iterating' over this queue.
+### Monitoring
 
+Resharding exposes a number of metrics and logs that allow for monitoring the resharding process as it is happening. Resharding requires manual recovery in case anything goes wrong and should be monitored in order to ensure smooth node operation. 
 
+* near_resharding_status is the primary metric that should be used for tracking the progress of resharding. It's tagged with a shard_uid label of the parent shard. It's set to corresponding ReshardingStatus enum and can take one of the following values
+  * 0 - Scheduled - resharding is scheduled and waiting to be executed. 
+  * 1 - Building - resharding is running. Only one shard at a time can be in that state while the rest will be either finished or waiting in the Scheduled state. 
+  * 2 - Finished - resharding is finished. 
+  * -1 - Failed - resharding failed and manual recovery action is required. The node will operate as usual until the end of the epoch but will then stop being able to process blocks. 
+* near_resharding_batch_size and near_resharding_batch_count - those two metrics show how much data has been resharded. Both should be gradually increasing while the near_resharding_status. 
+  * While in the Scheduled state both metrics should remain 0. 
+  * While in the Building state both metrics should be gradually increasing. 
+  * While in the Finished state both metrics should remain at the same value. 
+* near_resharding_batch_prepare_time_bucket, near_resharding_batch_apply_time_bucket and near_resharding_batch_commit_time_bucket - those three metrics can be used to track the performance of resharding and fine tune throttling if needed. As a rule of thumb the combined time of prepare, apply and commit for a batch should remain at the 100ms-200ms level on average. Higher batch processing time may lead to disruptions in block processing, missing chunks and blocks. 
```

Comment:
I would like some instructions on how to understand from metrics that resharding is going to finish before the start of the next epoch. In case we tuned too much into block processing optimisation. 
Maybe some approximations for `near_resharding_batch_size` or `near_resharding_batch_count` for testnet, mainnet, or something based on specific size of some columns (that is also a metric).

Reply:
Sure thing, I'll follow up in a separate PR.